src/platform/posix/posix_pollq_epoll.c

```diff
@@ -107,17 +112,19 @@ nni_posix_pfd_init(nni_posix_pfd **pfdp, int fd)
 	NNI_LIST_NODE_INIT(&pfd->node);
 
 	// notifications disabled to begin with
-	ev.events   = 0;
-	ev.data.ptr = pfd;
+	ev->events   = 0;
+	ev->data.ptr = pfd;
 
-	if (epoll_ctl(pq->epfd, EPOLL_CTL_ADD, fd, &ev) != 0) {
+	if (epoll_ctl(pq->epfd, EPOLL_CTL_ADD, fd, ev) != 0) {
```

Comment:
Could consider reading the result into a variable and free'ing `ev` right after before we start checking return codes ?